### PR TITLE
Updates query param minpipelinesversion to pipelinesversion in APIs

### PR DESCRIPTION
**Pull request by sm43**
Updates for both preview and v1 API and adds tests for regex check

This updates the query param as user would pass piplines version so
minPipelinesVersion would not make sense as we use it to refer what
version the task is compatible with, hence renames the param.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/v1/service/resource/resource.go [hunk 1]
```diff
@@ -205,9 +205,9 @@ func (s *service) ByCatalogKindName(ctx context.Context, p *resource.ByCatalogKi
 		}
 	}
 
-	// If minPipelinesVersion is passed then check for version compatible with pipelines version
-	if p.Minpipelinesversion != nil {
-		r = filterCompatibleVersions(r, *p.Minpipelinesversion)
+	// If pipelinesVersion is passed then check for version compatible with pipelines version
+	if p.Pipelinesversion != nil {
```
Comment (vinamra28) [inline api/v1/service/resource/resource.go:209]: shouldn't this be `p.PipelinesVersion` ?

Comment (sm43) [reply api/v1/service/resource/resource.go:209]: goa makes it title case `P`.
for `v` to be `V`, query param has to be changed from `pipelinesversion` to `pipelinesVersion` :)

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Comment by PuneetPunamiya** [comment]
@sm43  could you please rebase once

**Pull request merged**